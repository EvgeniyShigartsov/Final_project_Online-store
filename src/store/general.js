export const DOMAIN = 'https://intense-reaches-12245.herokuapp.com/api'
export default DOMAIN