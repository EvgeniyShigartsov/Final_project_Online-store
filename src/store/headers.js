export const headers = {
  Authorization: localStorage.getItem('token'),
  'Content-Type': 'application/json'
}

export const getHeaders = function () {
  return {
    Authorization: localStorage.getItem('token'),
    'Content-Type': 'application/json'
  }
}

export default headers;