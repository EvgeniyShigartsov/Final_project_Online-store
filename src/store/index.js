import { createStore, combineReducers, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import {
  persistStore,
  persistReducer
} from 'redux-persist'
import storage from 'redux-persist/lib/storage'
import {reducer as authReducer} from './auth/reducer';
import { reducer as productsReducer } from './products/reducer';

const persistConfig = {
  key: 'authLS',
  storage,
}

const persistedAuthReducer = persistReducer(persistConfig, authReducer)

const rootReducer = combineReducers({
  auth: persistedAuthReducer,
  products: productsReducer
});

export const store = createStore(rootReducer, composeWithDevTools(applyMiddleware(thunk)));
export const persistor = persistStore(store);

export default store;
