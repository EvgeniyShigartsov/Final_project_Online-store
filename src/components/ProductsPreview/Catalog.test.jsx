import React from 'react';
import { shallow } from 'enzyme';
import { render } from '@testing-library/react'

import { HashRouter as Router } from 'react-router-dom'
import { Provider } from 'react-redux'
import { store } from '../../store/index'

import { Catalog } from './Catalog'

describe('Catalog component', () => {
  test('should render Catalog component', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Router>
          <Catalog />
        </Router>
      </Provider>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})