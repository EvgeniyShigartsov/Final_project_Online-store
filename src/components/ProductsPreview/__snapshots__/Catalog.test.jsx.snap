// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catalog component should render Catalog component 1`] = `
<DocumentFragment>
  <div
    class="sc-hKgILt sc-kstrdz iUAErB gCZnHh"
  >
    <div
      class="sc-hBEYos hRhOZN"
    >
      <div
        class="sc-bdfBwQ cSCxSU"
        src="https://fuzzmusic.ru/wp-content/uploads/2020/08/pesni-iz-igry-kiberpank-2077-1.jpg"
      >
        <h3
          class="sc-gsTCUz iNRBaP"
        >
          gaming monitors
        </h3>
        <a
          class="sc-dlfnbm ddFrFr"
          href="#/catalog?categories=gamingMonitors"
        >
          View all gaming monitors
        </a>
      </div>
      <div
        class="sc-fodVxV loJcWr"
      />
    </div>
    <div
      class="sc-hBEYos hRhOZN"
    >
      <div
        class="sc-bdfBwQ ebNgOr"
        src="https://merchandising-assets.bestbuy.ca/bltc8653f66842bff7f/blt3ce6d340560cf08f/5e71863623ff3341c4f5aba8/computing-evergreen-feature-gaming-desktop-orientation-s.jpg?width=2877&quality=80"
      >
        <h3
          class="sc-gsTCUz iNRBaP"
        >
          desktops
        </h3>
        <a
          class="sc-dlfnbm ddFrFr"
          href="#/catalog?categories=desctops"
        >
          View all desktops
        </a>
      </div>
      <div
        class="sc-fodVxV loJcWr"
      />
    </div>
    <div
      class="sc-hBEYos hRhOZN"
    >
      <div
        class="sc-bdfBwQ eFJaIP"
        src="https://www.xmg.gg/wp-content/uploads/XMG_Pro-15_gg_Rotationsbanner_EN_Desktop.jpg"
      >
        <h3
          class="sc-gsTCUz iNRBaP"
        >
          laptops
        </h3>
        <a
          class="sc-dlfnbm ddFrFr"
          href="#/catalog?categories=laptops"
        >
          View all laptops
        </a>
      </div>
      <div
        class="sc-fodVxV loJcWr"
      />
    </div>
    <div
      class="sc-hBEYos hRhOZN"
    >
      <div
        class="sc-bdfBwQ kVUqRb"
        src="https://cdn.pixabay.com/photo/2015/06/24/15/45/hands-820272_1280.jpg"
      >
        <h3
          class="sc-gsTCUz iNRBaP"
        >
          tablets
        </h3>
        <a
          class="sc-dlfnbm ddFrFr"
          href="#/catalog?categories=tablets"
        >
          View all tablets
        </a>
      </div>
      <div
        class="sc-fodVxV loJcWr"
      />
    </div>
  </div>
</DocumentFragment>
`;
